examples/reason_react_router.re

```diff
@@ -0,0 +1,47 @@
+open ReactRouter;
+
+module HomePage = {
+    let component = ReasonReact.statelessComponent("HomePage");
+    let make = (_children) => {
+        ...component,
+        render: (_self) => 
+            <h1>{ReasonReact.stringToElement("Home page")}</h1>
+    }
+};
+
+module UserPage = {
+    let component = ReasonReact.statelessComponent("UserPage");
+    let make = (_children) => {
+        ...component,
+        render: (_self) => 
+            <h1>{ReasonReact.stringToElement("User page")}</h1>
+    }
+};
+
+module App = {
+    let component = ReasonReact.statelessComponent("App");
+    let blueStyle = ReactDOMRe.Style.make(~color="#000099", ());
+
+    let make = (_children) => {
+        ...component,
+        render: (_self) =>
+            <div>
+                <BrowserRouter>
+                    <div>
+                        <h1>{ReasonReact.stringToElement("Using NavLink")}</h1>
+                        <NavLink to_="/">{ReasonReact.stringToElement("Home")}</NavLink>
+                        <NavLink style=(blueStyle) to_="/user">{ReasonReact.stringToElement("User")}</NavLink>
+
+                        <h1>{ReasonReact.stringToElement("Using Link")}</h1>
+                        <Link to_="/">{ReasonReact.stringToElement("Home")}</Link>
+                        <Link to_="/user">{ReasonReact.stringToElement("User")}</Link>
+
+                        <Switch>
+                            <Route path="/" exact=true component=(() => <HomePage />) />
+                            <Route path="/user" component=(() => <UserPage />) />
```

Comment:
Suggestion: provide a comment that explain why you wrap the component into a function.

For example: https://github.com/emmenko/flopflip/blob/7007a4b9a2b0ab689edba20a1f4945b528154769/packages/reason-bindings/src/FlopFlip__Types.re#L3-L11

Reply:
Also be careful that if you keep it like this the `UserPage` component won't receive the router props (match, location, ...).

However we could do a little trick:

```reason
module UserPage = {
  let component = ...;
  let make = ...;
};
let userPageComponent =
  UserPage.(
    ReasonReact.wrapReasonForJs(~component, jsProps =>
      make(~match=jsProps##_match, ~location=jsProps##location, [||])
    )
  );

<Route component=userPageComponent />
```

Reply:
@emmenko The `wrapReasonForJs` returns a type of _ReactClass_ and the `component` attributes of the router requires a _ReactElement_ type. Should I use this method or transform another way ?

For the moment, I just fixed some comments you advised rebase on master.

Reply:
Ah right. Actually I think it's easier to use `ReactClass` (I also use it in one of my projects).

```reason
~component: option(ReasonReact.reactClass)=?,
```